deployer/hub.py

```diff
@@ -140,6 +142,46 @@ def auth_kubeconfig(self):
             os.environ['KUBECONFIG'] = decrypted_key_path
             yield
 
+    def auth_aws(self):
+        config = self.spec['aws']
+        key_path = config['key']
+        cluster_type = config['clusterType']
+        cluster_name = config['clusterName']
+        region = config['region']
+
+        with tempfile.NamedTemporaryFile() as kubeconfig:
+            orig_kubeconfig = os.environ.get('KUBECONFIG', None)
+            orig_access_key_id = os.environ.get('AWS_ACCESS_KEY_ID', None)
+            orig_secret_access_key = os.environ.get('AWS_SECRET_ACCESS_KEY', None)
+            try:
+                os.environ['KUBECONFIG'] = kubeconfig.name
+                with decrypt_file(key_path) as decrypted_key_path:
+
+                    decrypted_key_abspath = os.path.abspath(decrypted_key_path)
+                    if not os.path.isfile(decrypted_key_abspath):
+                        raise FileNotFoundError(
+                            'The decrypted key file does not exist')
+                    with open(decrypted_key_abspath) as f:
+                        creds = json.load(f)
+
+                    os.environ['AWS_ACCESS_KEY_ID'] = creds['AccessKeyId']
+                    os.environ['AWS_SECRET_ACCESS_KEY'] = creds['SecretAccessKey']
+
+                subprocess.check_call([
+                    'aws', 'eks', 'update-kubeconfig',
+                    f'--name={cluster_name}',
+                    f'--region={region}'
+                ])
+
```

Comment:
We could add support for kops-based clusters here with subprocessing `kops export kubecfg --admin` if the `cluster_type` is kops. We would need an additional parameter for the state store.

Reply:
What does it do? Does it consume AWS_ credentials to create a kubeconfig file?

I guess then that makes sense to me. Can we make it be written to the temp file allocated by python so we can benefit of having it automatically be deleted later then also?

Reply:
> What does it do? Does it consume AWS_ credentials to create a kubeconfig file?

Yep, `update-kubeconfig` in EKS (and `kops export kubecfg` on kops world) will create a new kubeconfig file (or append and existing one) modulo you have provided the required credentials (and have access to the state store in kops world).

> Can we make it be written to the temp file allocated by python so we can benefit of having it automatically be deleted later then also?

We are creating a temporary kubeconfig file in line 152, are you asking for something else?